src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -165,113 +166,91 @@ public MergeFeedsJob(Auth0UserProfile owner, Set<FeedVersion> feedVersions, Stri
-            regionalFeedSource = Persistence.feedSources.getById(project.regionalFeedSourceId);
-            // Create new feed source if this is the first regional merge.
-            if (regionalFeedSource == null) {
-                regionalFeedSource = new FeedSource("REGIONAL MERGE", project.id, REGIONAL_MERGE);
-                // Store new feed source.
-                Persistence.feedSources.create(regionalFeedSource);
-                // Update regional feed source ID on project.
-                project.regionalFeedSourceId = regionalFeedSource.id;
-                Persistence.projects.replace(project.id, project);
-            }
-        }
-        // Assign regional feed source or simply the first parent feed source found in the feed version list (these
-        // should all belong to the same feed source if the merge is not regional).
-        this.feedSource = mergeType.equals(REGIONAL)
-            ? regionalFeedSource
-            : feedVersions.iterator().next().parentFeedSource();
-        FeedRetrievalMethod retrievalMethod = mergeType.equals(REGIONAL)
-            ? REGIONAL_MERGE
-            : SERVICE_PERIOD_MERGE;
         // Merged version will be null if the new version should not be stored.
-        this.mergedVersion = storeNewVersion ? new FeedVersion(this.feedSource, retrievalMethod) : null;
+        this.mergedVersion = getMergedVersion(this, storeNewVersion);
         this.mergeFeedsResult = new MergeFeedsResult(mergeType);
     }
 
+    @BsonIgnore @JsonIgnore
+    public Set<FeedVersion> getFeedVersions() {
+        return this.feedVersions;
+    }
+
+    @BsonIgnore @JsonIgnore
+    public List<FeedToMerge> getFeedsToMerge() {
+        return this.feedsToMerge;
+    }
+
     /**
      * The final stage handles clean up (deleting temp file) and adding the next job to process the
      * new merged version (assuming the merge did not fail).
      */
+    @Override
     public void jobFinished() {
         // Delete temp file to ensure it does not cause storage bloat. Note: merged file has already been stored
         // permanently.
-        if (!mergedTempFile.delete()) {
-            // FIXME: send to bugsnag?
+        try {
+            Files.delete(mergedTempFile.toPath());
+        } catch (IOException e) {
             LOG.error(
                 "Merged feed file {} not deleted. This may contribute to storage space shortages.",
-                mergedTempFile.getAbsolutePath());
+                mergedTempFile.getAbsolutePath(),
+                e
+            );
+            ErrorUtils.reportToBugsnag(e, owner);
         }
     }
 
     /**
      * Primary job logic handles collecting and sorting versions, creating a merged table for all versions, and writing
      * the resulting zip file to storage.
      */
-    @Override public void jobLogic() throws IOException, CheckedAWSException {
+    @Override
+    public void jobLogic() throws IOException, CheckedAWSException {
```

Comment:
I think it'd be helpful to not have this method throw either of these exceptions so that finer-grained error handling can occur.